controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -297,7 +297,7 @@ func (r *KubeadmControlPlaneReconciler) updateStatus(ctx context.Context, kcp *c
 	kcp.Status.Replicas = replicas
 
 	remoteClient, err := r.remoteClientGetter(ctx, r.Client, util.ObjectKey(cluster), r.scheme)
-	if err != nil && !apierrors.IsNotFound(errors.Cause(err)) {
+	if err != nil && !apierrors.IsNotFound(errors.Cause(err)) && !apierrors.IsTimeout(errors.Cause(err)) {
```

Comment:
I'm wondering if we should short-circuit the logic below in the case that we get either of these acceptable errors? It doesn't seem like it would make much sense to attempt to use the client in `getMachineNode` in the case that we see either of these errors.

Reply:
yes, I'll fix it